src/smoke_simulator.cc

```diff
@@ -1,27 +1,109 @@
+#include <nanogui/nanogui.h>
 #include "smoke_simulator.hh"
 
-SmokeSimulator::SmokeSimulator(std::string shader_dir, nanogui::Screen *screen) {
+using namespace nanogui;
+
+SmokeSimulator::SmokeSimulator(std::string shader_dir, nanogui::Screen *screen, GLFWwindow *window)
+{
     shader = std::make_shared<nanogui::GLShader>();
     shader->initFromFiles("Default",
                           shader_dir + "/point.vert",
                           shader_dir + "/point.frag");
-    
-    for (int i = 0; i < 100; ++i) {
-        particles.emplace_back(Particle());
-    }
+
+    Vector3f init_velocity = Vector3f(0, 0.5, 0);
+    Vector4f init_color = Vector4f(66 / 256.0f, 135 / 256.0f, 245 / 256.0f, 1.0f);
+    emittor = new Emittor(Vector3f(0, -1, 0), init_velocity, init_color);
+    smoke = new Smoke();
```

Comment:
Why do we need pointer here? We can try to make `smoke` and `emitter` just regular member variable?